opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authorization/AuthorizationManager.java

```diff
@@ -81,244 +84,329 @@ static EnumSet<StudyPermissions.Permissions> getLockedAcls() {
+
+    void checkCanCreateUpdateDeleteVariableSets(String organizationId, long studyId, String userId) throws CatalogException;
+
+    default boolean isOpencgaAdministrator(JwtPayload payload) throws CatalogException {
+        return isOpencgaAdministrator(payload.getOrganization(), payload.getUserId());
+    }
 
-    void checkCanCreateUpdateDeleteVariableSets(long studyId, String userId) throws CatalogException;
+    boolean isOpencgaAdministrator(String organization, String userId) throws CatalogException;
 
-    Boolean isInstallationAdministrator(String user);
+    default void checkIsOpencgaAdministrator(JwtPayload payload) throws CatalogException {
+        checkIsOpencgaAdministrator(payload, null);
+    }
 
-    void checkIsInstallationAdministrator(String user) throws CatalogException;
+    default void checkIsOpencgaAdministrator(JwtPayload payload, String action) throws CatalogException {
+        checkIsOpencgaAdministrator(payload.getOrganization(), payload.getUserId(), action);
+    }
 
-    void checkIsOwnerOrAdmin(long studyId, String userId) throws CatalogException;
+    default void checkIsOpencgaAdministrator(String organization, String userId) throws CatalogException {
+        checkIsOpencgaAdministrator(organization, userId, null);
+    }
 
-    Boolean isOwnerOrAdmin(long studyId, String userId) throws CatalogException;
+    void checkIsOpencgaAdministrator(String organization, String userId, String action) throws CatalogException;
 
-    void checkFilePermission(long studyId, long fileId, String userId, FilePermissions permission) throws CatalogException;
+    /**
+     * Check if the given user is the owner of the organization or if it is an admin.
+     * It does not include the opencga admins.
+     * If the user is not the owner or an admin, it will throw an exception.
+     *
+     * @param organizationId    Organization id
+     * @param userId            User id
+     * @throws CatalogException CatalogException
+     */
+    void checkIsOrganizationOwnerOrAdmin(String organizationId, String userId) throws CatalogException;
 
-    void checkSamplePermission(long studyId, long sampleId, String userId, SamplePermissions permission)
+    /**
+     * Check if the given user is the owner of the organization or if it is an admin.
+     * It does not include opencga admins.
+     *
+     * @param organization Organization
+     * @param userId User id
+     * @return true if the user is the owner or an admin of the organization
+     * @throws CatalogDBException CatalogDBException
+     */
+    boolean isOrganizationOwnerOrAdmin(String organization, String userId) throws CatalogDBException;
+
+    /**
+     * Check if the user is part of the {@link ParamConstants#ADMINS_GROUP} group of the study.
+     * Keep in mind that all organization admins and the organization owner are also study admins.
+     * It does not include the opencga admins.
```

Comment:
isStudyAdmin includes
- studyAdmin
- organizationAdmin
- organizationOwner

but excludes "opencgaAdmin" .. why? Same as before, we have a lot of queries to both methods:
```java
!isOpencgaAdministrator(organizationId, userId) && !isStudyAdministrator(organizationId, studyId, userId) 
```

Reply:
I understand that there is nothing that an opencgaAdministrator shouldn't be allowed to do

Reply:
Agree. To be changed in the next release.